### ISSUE TITLE
[Feature Request] Use plugin from service

### ISSUE THREAD

**Issue by Nico04** [michael-cheung-thebus/foreground_service#10]
It would be great we could use Flutter plugins (with native part) from the service.
For now, we can't even get this plugin from the service, to be able to update the notification for instance. For instance, this line return a MissingPluginException
`final fgsIsRunning = await ForegroundService.foregroundServiceIsStarted();`

Maybe this plugin can help :
https://pub.dev/packages/flutter_isolate

**Comment by MbarkErras** [comment]
Hi @Nico04 is there anyway to use foreground_service with flutter_isolate to achieve this?

**Comment by Nico04** [comment]
I don't know, I had to give up this subjet unfortunately, sorry

**Comment by michael-cheung-thebus** [comment]
I've modified it so that the line in the original post will now function properly, but unfortunately when it comes to using plugins in general, this appears to be a limitation inherent to the current state of Dart/Flutter.

The service function runs in an isolate, and isolates are currently unable to/not supported for using plugins: https://github.com/flutter/flutter/issues/13937

There was a short-lived attempt to try and work around this, but it looks like it's not maintaned/doesn't work anymore and at any rate would not have worked with how this plugin needs to function: https://pub.dev/packages/flutter_isolate

In short, AFAICT, this feature is impossible until further notice.

**Comment by Nico04** [comment]
I understand, thank you for your answer.
Did you have a look at this plugin ?
https://pub.dev/packages/audio_service

It's very interesting as it does that : executing Dart code in a background service, communicating with main thread.
Maybe it can help ?
